src/main.rs

```diff
@@ -27,9 +30,44 @@ fn main() {
     let block_size = 1024;
 
     // Read audio data and write it to the output text file (one column per channel)
-    let mut out = File::create(&args[2]).expect("Unable to create file");
+    //let mut out = File::create(&args[2]).expect("Unable to create file");
+    let mut out = hound::WavWriter::create(&args[2], spec).unwrap();
+    let mut sample_index = 0;
+    let mut block_index: u32 = 0;
+    let mut input_buffer = vec![vec![f32::default(); channels as usize]; block_size as usize];
+    let mut output_buffer = vec![vec![f32::default(); channels as usize]; block_size as usize];
+    let mut comb_filter = CombFilter::new(FilterType::FIR, 1.0, 44100.0, channels as usize);
     for (i, sample) in reader.samples::<i16>().enumerate() {
         let sample = sample.unwrap() as f32 / (1 << 15) as f32;
-        write!(out, "{}{}", sample, if i % channels as usize == (channels - 1).into() { "\n" } else { " " }).unwrap();
+        let channel_index = i % channels as usize;
+        if sample_index < block_size {
+            input_buffer[sample_index][channel_index] = sample;
+        } else {
+            let input_buffer_slice = &input_buffer[..].iter().map(|v| v.as_slice()).collect::<Vec<_>>();
+            let input_buffer_slice_2d = &input_buffer_slice[..];
+            let mut output_buffer_slice = output_buffer[..].iter_mut().map(|v| v.as_mut_slice()).collect::<Vec<_>>();
+            let output_buffer_slice_2d = &mut output_buffer_slice[..];
+            comb_filter.process(input_buffer_slice_2d, output_buffer_slice_2d);
+            for block_index in output_buffer_slice_2d.iter_mut() {
```

Comment:
I'm wondering how you handled the edges where the delayed samples flow into the next block

Reply:
This could be the source of some error in the plot. Let me check and see if I'm understanding it correctly.

Reply:
I looked over the code and all the delayed samples are stored in ring buffers for each channel which are part of the `comb_filter.rs` file. That way they are separate from the input and output buffers defined here and still remembered at the next block.